cnc_ctrl_v1/Axis.cpp

```diff
@@ -103,10 +105,32 @@ void   Axis::computePID(){
     if (_pidController.Compute()){
         // Only write output if the PID calculation was performed
         motorGearboxEncoder.write(_pidOutput);
-    }
+
+        if(_pidOutput) {
+            if (millis() - stallmillis > STALLTIMEOUT)
+            {
+                stallmillis = millis();
+                
+                long delta=lastcoder - motorGearboxEncoder.encoder.read(); 
+                
+                if(lastcoder == 0 || delta >  STALLSTEPS || delta < STALLSTEPS)
+                {
+                    moved=lastcoder !=0;
```

Comment:
I don't understand this.  Is this a form of a ternary I wasn't aware of does it read:

set moved to equal lastcoder if lastcoder is not 0?

If so, this is a problem.  moved is a char and lastcoder is a long.  This will almost certainly overflow.

Reply:
No, it's set moved to `if lastcoder is not zero` (so if lastcoder=0 -- the initial run of the function (otherwise it'll be the last encoder value), then moved=0.  OTOH, if delta is outside of STALLSTEPS, then it /has moved/.